pkg/api/v1alpha1/condition_consts.go

```diff
@@ -0,0 +1,49 @@
+package v1alpha1
+
+// Conditions and condition Reasons for the Cluster object.
+const (
+	// ReadyCondition reports a summary of other conditions, indicating an overall operational
+	// state of the cluster.
+	ReadyCondition ConditionType = "Ready"
+
+	// PendingUpdateReason reports when an update has been scheduled but has yet to happen.
+	PendingUpdateReason = "PendingUpdate"
+
+	// ControlPlaneInitializedCondition reports that all the control plane nodes are
+	// in a fully ready and running state.
+	ControlPlaneReadyCondition ConditionType = "ControlPlaneReady"
+
+	// ControlPlaneInitializedCondition reports that the first control plane instance has been initialized
+	// and so the control plane is available and an API server instance is ready for processing requests.
+	ControlPlaneInitializedCondition ConditionType = "ControlPlaneInitialized"
+
+	// WaitingForControlPlaneInitializedReason used when cluster is waiting for control plane to be initialized.
+	WaitingForControlPlaneInitializedReason = "WaitingForControlPlaneInitialized"
+
+	// FirstControlPlaneUnavailableMessage used when waiting for the first control plane instance to become
+	// initialized and available.
+	FirstControlPlaneUnavailableMessage = "The first control plane instance is not available yet"
+
+	// WaitingForControlPlaneReadyReason used when cluster is waiting for control plane nodes to transition to a
+	// fully ready and running state.
+	WaitingForControlPlaneReadyReason = "WaitingForControlPlaneReady"
```

Comment:
same here, this doesn't say anything
what we want is the actual reasons why is not ready:
- init in progress
- scaling up
- scaling down
- upgrade in process
...

Reply:
Okay. I think I will keep this PR focused for now by removing these reasons then include them in subsequent PRs when they're used